src/Bicep.Core/Semantics/SemanticModel.cs

```diff
@@ -114,14 +115,85 @@ public bool HasErrors()
         /// a symbol will always be returned. Binding failures are represented with a non-null error symbol.
         /// </summary>
         /// <param name="syntax">the syntax node</param>
-        public Symbol? GetSymbolInfo(SyntaxBase syntax) => this.Binder.GetSymbolInfo(syntax);
+        public Symbol? GetSymbolInfo(SyntaxBase syntax)
+        {
+            static PropertySymbol? GetPropertySymbol(TypeSymbol? baseType, string property)
+            {
+                var bodyObjectType = baseType switch
```

Comment:
>switch [](start = 46, length = 6)

Can you reuse `TypeAssignmentVisitor.UnwrapType()` here?

Reply:
Yup! I looked for that function but didn't find it.